src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -785,63 +787,71 @@ internal async Task InvokeResponse(InvocationResponse invokeResponse)
-                    {
-                        Outputs = bindingsDictionary,
-                        Return = invokeResponse?.ReturnValue?.ToObject()
-                    };
-                    context.ResultSource.SetResult(result);
-                }
-                catch (Exception responseEx)
-                {
-                    context.ResultSource.TrySetException(responseEx);
-                }
-                finally
-                {
-                    // Free memory allocated by the host (for input bindings) which was needed only for the duration of this invocation
-                    if (!_sharedMemoryManager.TryFreeSharedMemoryMapsForInvocation(invokeResponse.InvocationId))
+                        var result = new ScriptInvocationResult()
+                        {
+                            Outputs = bindingsDictionary,
+                            Return = invokeResponse?.ReturnValue?.ToObject()
+                        };
+                        context.ResultSource.SetResult(result);
+                    }
+                    catch (Exception responseEx)
                     {
-                        _workerChannelLogger.LogWarning($"Cannot free all shared memory resources for invocation: {invokeResponse.InvocationId}");
+                        context.ResultSource.TrySetException(responseEx);
                     }
-
-                    // List of shared memory maps that were produced by the worker (for output bindings)
-                    IList<string> outputMaps = GetOutputMaps(invokeResponse.OutputData);
-                    if (outputMaps.Count > 0)
+                    finally
                     {
-                        // If this invocation was using any shared memory maps produced by the worker, close them to free memory
-                        SendCloseSharedMemoryResourcesForInvocationRequest(outputMaps);
+                        // Free memory allocated by the host (for input bindings) which was needed only for the duration of this invocation
+                        if (!_sharedMemoryManager.TryFreeSharedMemoryMapsForInvocation(invokeResponse.InvocationId))
+                        {
+                            _workerChannelLogger.LogWarning($"Cannot free all shared memory resources for invocation: {invokeResponse.InvocationId}");
+                        }
+
+                        // List of shared memory maps that were produced by the worker (for output bindings)
+                        IList<string> outputMaps = GetOutputMaps(invokeResponse.OutputData);
+                        if (outputMaps.Count > 0)
+                        {
+                            // If this invocation was using any shared memory maps produced by the worker, close them to free memory
+                            SendCloseSharedMemoryResourcesForInvocationRequest(outputMaps);
+                        }
                     }
                 }
             }
+            else
```

Comment:
Looks like this else is at the wrong level - it should be for "if (invokeResponse.Result.IsInvocationSuccess"

Reply:
There should be a an existing test somewhere that we can add to to ensure these new metrics are emitted correctly. I'm assuming we have tests for worker invocations already.